docs/rpc-endpoints.md

```diff
@@ -0,0 +1,255 @@
+
+Fetch the contract interface for a given contract, identified by [Stacks Address] and [Contract Name].
+
+This returns a JSON object of the form:
+
+```
+{
+  "functions": [
+    {
+      "name": "exotic-block-height",
+      "access": "private",
+      "args": [
+        {
+          "name": "height",
+          "type": "uint128"
+        }
+      ],
+      "outputs": {
+        "type": "bool"
+      }
+    },
+    {
+      "name": "update-info",
+      "access": "public",
+      "args": [],
+      "outputs": {
+        "type": {
+          "response": {
+            "ok": "bool",
+            "error": "none"
+          }
+        }
+      }
+    },
+    {
+      "name": "get-exotic-data-info",
+      "access": "read_only",
+      "args": [
+        {
+          "name": "height",
+          "type": "uint128"
+        }
+      ],
+      "outputs": {
+        "type": {
+          "tuple": [
+            {
+              "name": "btc-hash",
+              "type": {
+                "buffer": {
+                  "length": 32
+                }
+              }
+            },
+            {
+              "name": "burn-block-time",
+              "type": "uint128"
+            },
+            {
+              "name": "id-hash",
+              "type": {
+                "buffer": {
+                  "length": 32
+                }
+              }
+            },
+            {
+              "name": "stacks-hash",
+              "type": {
+                "buffer": {
+                  "length": 32
+                }
+              }
+            },
+            {
+              "name": "stacks-miner",
+              "type": "principal"
+            },
+            {
+              "name": "vrf-seed",
+              "type": {
+                "buffer": {
+                  "length": 32
+                }
+              }
+            }
+          ]
+        }
+      }
+    }
+  ],
+  "variables": [],
+  "maps": [
+    {
+      "name": "block-data",
+      "key": [
+        {
+          "name": "height",
+          "type": "uint128"
+        }
+      ],
+      "value": [
+        {
+          "name": "btc-hash",
+          "type": {
+            "buffer": {
+              "length": 32
+            }
+          }
+        },
+        {
+          "name": "burn-block-time",
+          "type": "uint128"
+        },
+        {
+          "name": "id-hash",
+          "type": {
+            "buffer": {
+              "length": 32
+            }
+          }
+        },
+        {
+          "name": "stacks-hash",
+          "type": {
+            "buffer": {
+              "length": 32
+            }
+          }
+        },
+        {
+          "name": "stacks-miner",
+          "type": "principal"
+        },
+        {
+          "name": "vrf-seed",
+          "type": {
+            "buffer": {
+              "length": 32
+            }
+          }
+        }
+      ]
+    }
+  ],
+  "fungible_tokens": [],
```

Comment:
Are we camelCasing of snake_casing our fields?

Reply:
I think camelCase -- but I didn't want to mess with the Contract ABI too much in this PR, because it's being consumed elsewhere as well, so I'd prefer if that was a separate PR.

Reply:
FWIU most json APIs use snake_case
src
* https://stripe.com/docs/api/expanding_objects
* https://developer.github.com/v3/issues/
* https://developers.coinbase.com/api/v2#users

Reply:
Okay -- we can change to snake_case instead.